README.md

```diff
@@ -1,199 +1,145 @@
+* `factor`: the factor definining the low-resolution grid of the generated data
+  with respect to the high-resolution grid.
+* `CO2`: 0 for control, 1 for 1% increase per year dataset.
+* `global`: TODO
 
-# Run Testing code
-Testing is carried out in an interactive sbatch run, where we ask for a GPU:
+Direct call (without MLflow) example:
 
-```
-srun -t1:30:00 --mem=40000 --gres=gpu:1 --pty /bin/bash
-```
+    python src/gz21_ocean_momentum/cmip26.py -85 85 -280 80 --factor 4 --ntimes 10
 
-We then start our singularity container and activate the subgrid conda environment:
+MLflow call example:
 
-- Start a singularity container with the appropriate image via the following command:
 ```
-singularity exec --nv --overlay /scratch/ag7531/overlay2 /scratch/work/public/singularity/cuda11.0-cudnn8-devel-ubuntu18.04.sif /bin/bash
-```
-- Activate conda:
-```
-source /ext3/env.sh
-```
-- Activate the subgrid conda environment:
-```
-conda activate subgrid
+mlflow run . --experiment-name <name>--env-manager=local \
+-P lat_min=-25 -P lat_max=25 -P long_min=-280 -P long_max=80 \
+-P factor=4 \
+-P CO2=1 -P global=0 \
+-P ntimes=100 \
+-P chunk_size=1
 ```
 
-We then cd to /home/ag7531/code/subgrid and run:
+#### Training
+MLflow call example:
 
 ```
-python -m testing.main.py --n_splits 50 --batch_size 2 --n_test_times 7300
+mlflow run . --experiment-name raehik -e train --env-manager=local \
+-P exp_id=692154129919725696 -P run_id=c57b36da385e4fc4a967e7790192ecb2 \
+-P learning_rate=0/5e-4/15/5e-5/30/5e-6 -P n_epochs=200 -P weight_decay=0.00 -P train_split=0.8 \
+-P test_split=0.85 -P model_module_name=models.models1 -P model_cls_name=FullyCNN -P batchsize=4 \
+-P transformation_cls_name=SoftPlusTransform -P submodel=transform3 \
+-P loss_cls_name=HeteroskedasticGaussianLossV2
 ```
 
+Relevant parameters:
 
+* `exp_id`: id of the experiment containing the run that generated the forcing
+  data.
+* `run_id`: id of the run that generated the forcing data that will be used for
+  training.
+* `loss_cls_name`: name of the class that defines the loss. This class should be
+  defined in train/losses.py in order for the script to find it. Currently the
+  main available options are:
+  * `HeteroskedasticGaussianLossV2`: this corresponds to the loss used in the
+    paper
+  * `BimodalGaussianLoss`: a Gaussian loss defined using two Gaussian modes
+* `model_module_name`: name of the module that contains the class defining the
+  NN used
+* `model_cls_name`: name of the class defining the NN used, should be defined in
+  the module specified by model_module_name
 
+Another important way to modify the way the script runs consists in modifying
+the domains used for training. These are defined in `training_subdomain.yaml` in
+terms of their coordinates. Note that at run time domains will be truncated to
+the size of the smallest domain in terms of number of points.
```

Comment:
It would be useful if some information could be added about data size and run times, and flag if and when more capacity than provided by a standard laptop/desktop computer is needed. I thought I also saw something somewhere about GPU usage, but I might misremember or confuse this with something else.

Reply:
I don't have enough info on resource usage currently, nor do I know how to estimate from e.g. Pytorch model size. Should become more apparent when we're further along.